centreon/tests/php/Centreon/Application/Validation/Constraints/UniqueEntityTest.php

```
@@ -0,0 +1,307 @@
+
+        $value = new class () {
+            public function getName(): string { return 'foo'; }
+        };
+
+        $this->validator->validate($value, new UniqueEntity(field: 'name', repository: 'SomeRepo'));
+    }
+
+    public function testNoResultSkipsValidation(): void
+    {
+        $repository = new class () {
+            public function findOneBy(array $criteria): mixed { return null; }
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$criteria'. (undefined)

(UnusedFormalParameter)